File: src/main/java/com/hackerrank/introduction/javaloops2/Solution.java
package com.hackerrank.introduction.javaloops2;

import java.util.ArrayList;
import java.util.Scanner;

public class Solution {

  public static void main(String[] args) {
    final Scanner in = new Scanner(System.in);
    final int t = in.nextInt();

    final ArrayList<Integer> series = new ArrayList<>();

    for (int i = 0; i < t; i++) {
      series.clear();

      final int a = in.nextInt();
      final int b = in.nextInt();
      final int n = in.nextInt();

      for (int exp = 0; exp < n; exp++) {
        final int term = expansion(a, b, exp);
        series.add(term);
      }

      for (int term : series) {
        System.out.printf("%d ", term);
      }
      System.out.println();
    }

    in.close();
  }

  private static int expansion(int a, int b, int exp) {
    if (exp < 1) {
      return a + b;
    }
    return (int) Math.pow(2, exp) * b + expansion(a, b, exp - 1);
  }
}

File: src/main/kotlin/com/hackerrank/introduction/javaloops2/Solution.kt
package com.hackerrank.introduction.javaloops2

import java.util.*

fun main(args: Array<String>) {
  val `in` = Scanner(System.`in`)
  val t = `in`.nextInt()

  val series = ArrayList<Int>()

  for (i in 0..t - 1) {
    series.clear()

    val a = `in`.nextInt()
    val b = `in`.nextInt()
    val n = `in`.nextInt()

    for (exp in 0..n - 1) {
      val term = expansion(a, b, exp)
      series.add(term)
    }

    for (term in series) {
      System.out.printf("%d ", term)
    }
    println()
  }

  `in`.close()
}

private fun expansion(a: Int, b: Int, exp: Int): Int {
  if (exp < 1) {
    return a + b
  }
  return Math.pow(2.0, exp.toDouble()).toInt() * b + expansion(a, b, exp - 1)
}
